Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m

```diff
@@ -76,21 +119,79 @@ - (void)verifyPhoneNumber:(NSString *)phoneNumber
         });
       }
     };
+    [self internalVerifyPhoneNumber:phoneNumber completion:callBackOnMainThread];
+  });
+}
 
-    if (!phoneNumber.length) {
-      callBackOnMainThread(nil,
-                           [FIRAuthErrorUtils missingPhoneNumberErrorWithMessage:nil]);
-      return;
-    }
-    [_auth.notificationManager checkNotificationForwardingWithCallback:
-        ^(BOOL isNotificationBeingForwarded) {
-      if (!isNotificationBeingForwarded) {
-        callBackOnMainThread(nil, [FIRAuthErrorUtils notificationNotForwardedError]);
+- (void)verifyPhoneNumber:(NSString *)phoneNumber
+               UIDelegate:(nullable id<FIRAuthUIDelegate>)UIDelegate
+               completion:(nullable FIRVerificationResultCallback)completion {
+  dispatch_async(FIRAuthGlobalWorkQueue(), ^{
+    FIRVerificationResultCallback callBackOnMainThread = ^(NSString *_Nullable verificationID,
+                                                           NSError *_Nullable error) {
+      if (completion) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+          completion(verificationID, error);
+        });
+      }
+    };
+    [self internalVerifyPhoneNumber:phoneNumber completion:^(NSString *_Nullable verificationID,
+                                                             NSError *_Nullable error) {
+      if (!error) {
+        // Associate the phone number with the verification ID.
+        verificationID.fir_authPhoneNumber = phoneNumber;
```

Comment:
Hasn't `fir_authPhoneNumber` already assigned in `verifyClientAndSendVerificationCodeToPhoneNumber:`?

Reply:
Yes, removed.

Reply:
Removing the assignment from `verifyClientAndSendVerificationCodeToPhoneNumber:...` affects the existing `verifyPhoneNumber:completion:` method. Why would we do that? It seems the assignment here should be removed instead.

Reply:
It was not removed from`verifyClientAndSendVerificationCodeToPhoneNumber:...` but rather from the end of this method, this was fixed.  This assignment however doesn't seem to be used at all anymore and will likely be removed in a follow up PR.